### PR TITLE
Reintroduce pause

### PR DESCRIPTION
**Pull request by breunigs**
not yet tested

### CODE REVIEW

**Comment by breunigs** [comment]
depends on https://github.com/appsignal/rdkafka-ruby/pull/77 (and a new release, I guess)

File: README.md [hunk 1]
```diff
@@ -155,23 +155,11 @@ Any headers set on the message will be available when consuming the message:
...
-
-  # Signals back to Kafka that we're still alive!
-  heartbeat
-
-  long_running_op_two(message)
-end
-```
+If you have tasks that take longer than 5 minutes, you'll need to adjust `max_poll_interval`. If your consumer takes longer than this value to fetch the next message from Kafka a group rebalance will occur. Note that this applies even when batch processing, so you might want to adjust the batch size with `fetch_messages`. The default is 5 minutes.
 
+In order to detect if your consumer died earlier than the `max_poll_interval` heartbeats will be sent regularly in the background. This happens on a separate thread and connection, so that your workload doesn't starve the heartbeats. You can configure when heartsbeats are sent exactly using `session_timeout` and `heartbeat_interval` options. Usually the default values are fine and don't need adjustment.
```
Comment (JHK) [inline README.md]: Some points I noticed:
 - Describe the rationale first. I think `Heartbeats` had a good start. See also next point
 - Why does a rebalance occur when my processing takes too long?
 - Default values are meant to be fine, that was the one of the reasonings for racecar over direct kafka lib usage
 - _5 minutes_ is mentioned trice (once as _this value_)

File: lib/racecar/consumer.rb [hunk 2]
```diff
@@ -63,7 +63,7 @@ def produce(topic:, payload:, key:, headers: nil)
     end
 
     def heartbeat
-      raise "not yet implemented for rdkafka"
+      warn "DEPRECATION WARNING: Manual heartbeats are not supported and not needed with librdkafka."
```
Comment (JHK) [inline lib/racecar/consumer.rb:66]: In general 👍 
But we need to clarify with upstream how to handle breaking changes. I'd suggest to add those warnings in the current version and just remove this method.

File: CHANGELOG.md [hunk 3]
```diff
@@ -23,10 +23,8 @@
 * [Instrumentation] `acknowledged_message.racecar` send whenever a produced message was successfully received by Kafka. Payload includes `offset` and `partition`, but no message details.
 * [Instrumentation] `rdkafka-ruby` does not yet provide instrumentation [rdkafka-ruby#54](https://github.com/appsignal/rdkafka-ruby/issues/54)
 * [Instrumentation] if processors define a `statistics_callback`, it will be called once every second for every subscription or producer connection. The first argument will be a Hash, for contents see [librdkafka STATISTICS.md](https://github.com/edenhill/librdkafka/blob/master/STATISTICS.md)
-
 * Add current directory to `$LOAD_PATH` only when `--require` option is used (#117).
-
-* TODO: reintroduce manual heartbeat support
+* Remove manual heartbeat support, as it's not required with librdkafka. If you run into rebalancing during long tasks, either reduce the batch size (`fetch_messages`) or increase the maximum time between messages fetches (`max_poll_interval`).
```
Comment (JHK) [inline CHANGELOG.md]: We should also refer to the changed manual paragraph here.

File: lib/racecar/consumer.rb (hunk 2 above)
Comment (breunigs) [reply lib/racecar/consumer.rb:66]: Agreed. I intentionally kept it here because it's easier to spot than the breaking change of removing it.

**Review by JHK — APPROVED** [review]
Lets wait for the other change to be merged.

**Pull request merged**